backends/gdx-backend-lwjgl3/src/com/badlogic/gdx/backends/lwjgl3/DefaultLwjgl3Input.java

```diff
@@ -301,9 +323,79 @@ public void getTextInput (TextInputListener listener, String title, String text,
 	}
 
 	@Override
-	public void getTextInput (TextInputListener listener, String title, String text, String hint, OnscreenKeyboardType type) {
-		// FIXME getTextInput does nothing
-		listener.canceled();
+	public void getTextInput (final TextInputListener listener, String title, String text, String hint,
+		OnscreenKeyboardType type) {
+		// use swing fixit, by deedywu
+		SwingUtilities.invokeLater(new Runnable() {
+			@Override
+			public void run () {
+				final JFrame frame = new JFrame(title);
+				JPanel panel = new JPanel();
+				// default 30 char length
+				JTextField textField;
+				switch (type) {
+				case Password:
+					textField = new JPasswordField(text, 30); // 用星号来隐藏密码
+					((JPasswordField)textField).setEchoChar('*');
+					break;
+				case NumberPad:
+					NumberFormat numberFormat = NumberFormat.getNumberInstance(Locale.getDefault());
+					DecimalFormat decimalFormat = (DecimalFormat)numberFormat;
+					decimalFormat.setGroupingUsed(false);
+					textField = new JFormattedTextField(decimalFormat);
+					textField.setText(text);
+					textField.setColumns(30);
+					textField.setDocument(new PlainDocumentNumber());
+					break;
+				default:
+					textField = new JTextField(text, 30);
+					break;
+				}
+				textField.setFocusable(true);
+				textField.setToolTipText(hint);
+				JButton button = new JButton("OK");
+				Container contentPane = frame.getContentPane();
+				contentPane.setLayout(new BorderLayout());
+				panel.add(textField);
+				panel.add(button);
+				contentPane.add(panel);
+				frame.pack();
+				frame.setLocationRelativeTo(null);
+				frame.setVisible(true);
+				// enter ok
+				frame.getRootPane().setDefaultButton(button);
+				// esc cancel
+				frame.getRootPane().registerKeyboardAction(new ActionListener() {
+					@Override
+					public void actionPerformed (ActionEvent actionEvent) { // on cancel
+						listener.canceled();
+						frame.dispose();
+					}
+				}, KeyStroke.getKeyStroke(KeyEvent.VK_ESCAPE, 0), JComponent.WHEN_IN_FOCUSED_WINDOW);
+				button.addActionListener(new ActionListener() {
+					@Override
+					public void actionPerformed (ActionEvent e) { // on ok
+						listener.input(textField.getText());
+						frame.dispose();
+					}
+				});
+			}
+		});
+	}
+
+	/*** input number filter */
+	private static class PlainDocumentNumber extends PlainDocument {
+		@Override
+		public void insertString (int offset, String s, AttributeSet attrSet) throws BadLocationException {
```

Comment:
Wouldn't this implementation disallow inputting negative numbers?

Reply:
As well as floating-point numbers, because of the `'.'`

Reply:
Haha, this is indeed a bug, because my program is more restrictive, sorry, I will fix it for general purpose now